### ISSUE TITLE
Workflow nodejs-ci.yml does not fail when PR is out of date with main trunk

### ISSUE THREAD

**Issue by WoodyB** [WoodyB/vending-machine-project#411]
**Scenario 1**
Developer is working on a code change locally. Meanwhile the main branch (trunk) has some updates in the same file he/she is working on but the developer isn't aware of this.

**Scenario 2**
Developer is working on a code change locally but forgot to sync with the main trunk from the last change so that they are one commit behind main

In both scenarios the developer pushes their changes up to GitHub then creates a PR. This triggers the nodejs-ci.yml workflow.

**Expected Outcome**
The workflow should fail alerting the developer of the issue

**Actual Outcome**
The merge fails and can be detected in the log file from the workflow.

See https://github.com/WoodyB/vending-machine-project/actions/runs/8149480543

**Comment by WoodyB** [comment]
I attempted a fix but proved it did not work. The good news is I know exactly how to replicate this bug.
Step 1: Make a small change to the ReadMe file in a feature branch
Step 2: Submit a PR. This PR will trigger the nodejs-ci.yml as expected
Step 3: Once the change has been merged to main return to your local machine and change back to you local main branch (BUT DO NOT PULL) we are purposely going to create another PR without doing this.
Step 4 From your local out of sync main branch create a new feature branch git checkout -b <new branch>
Step 5 Now make a small change in the ReadMe file (BUT NOT ON THE SAME LINE) If the change is on the same line GitHub may be able to detect the merge conflict and never not run the workflow. What we want is to be out of sync, missing some changes.
Step 6 Commit you change to your local branch then git push origin <out of sync branch>
Step 7: Now goto GitHub and create the PR which will trigger the workflow.

Observe: If you were successful with creating the scenario you will see 
![image](https://github.com/WoodyB/vending-machine-project/assets/42235/87c1042e-658f-42ac-8bc7-03e3787c6ffc)

In this scenario the workflow will be triggered.

Expected Result:
Workflow should detect this issue and fail. Suggesting the developer makes sure their branch is not out of sync with main. The PR should remain open and the branch should not be deleted.

Actual Result:
The workflow does not fail. The merge fails but the workflow does not detect it. The branch is deleted by the workflow. The poor developer will wonder why their code change was never merged because from their point of view it was. Arguments will breakout and possibly results in fistfights that could result in deaths.
